odoo/tools/template_inheritance.py

```diff
@@ -217,6 +217,28 @@ def extract(spec):
                     if child.get('position') == 'move':
                         child = extract(child)
                     node.addprevious(child)
+            elif pos == 'content':
+                ''' Replace the entire content of an element '''
+                target = copy.deepcopy(node)
+                # Clear the content and attributes
+                node.clear()
+                # Get back attributes
+                for attr in target.keys():
+                    node.set(attr, target.get(attr))
+
+                # Add xpath content
+                if spec.text is not None:
+                    node.text = spec.text.lstrip()
+                for child in spec.getchildren():
+                    node.append(child)
+
+                # If the new content has children, right strip the tail of the last one
+                childs = node.getchildren()
+                if len(childs) > 0 and childs[-1].tail is not None:
+                    childs[-1].tail = childs[-1].tail.rstrip()
+                # Otherwise right strip the text
+                elif spec.text is not None:
+                    node.text = node.text.rstrip()
```

Comment:
Are you sure you need that many lines of code to implement this ? Is not the code just about removing the children and text of the current node and adding the new children ? I don't understand why there is manipulation of tail and attributes.

Reply:
So, node is the target ans spec is the xpath. If I want to append spec into node, it will add xpath node with it.
Getting children of spec only give me others tag elements with their content.
```
<xpath ...>
    new <span> content </span> to add
</xpath>
```
The child of this xpath is the <span> which has a tail "to add". Adding this child to node will result to : `<span> content </span> to add`
to get the element before the first tag, I need to get the text content of my xpath (spec), and adding it to the node's text value.
Doing that creates some weird spaces at the start and the end of node. So, I use strip to correct it.
I'll try to figure out a way to improve that !